### DISCUSSION TITLE
How can i stlye a Link in styled components

### DISCUSSION THREAD

**Discussion by deki84** [neuefische/questions#196] [Web] [answered]
Hello, I have been looking for a long time how to stly links to styled components. Can any of you help me? Thanks in advance.

<img width="492" alt="Screenshot 2022-09-30 at 11 45 40" src="https://user-images.githubusercontent.com/35610295/193243212-8a693224-b803-4f45-930a-71677272cbf8.png">
<img width="478" alt="Screenshot 2022-09-30 at 11 46 07" src="https://user-images.githubusercontent.com/35610295/193243294-c6afed1b-c7a9-430d-9bab-257b4b322257.png">
<img width="503" alt="Screenshot 2022-09-30 at 11 46 50" src="https://user-images.githubusercontent.com/35610295/193243543-c30ad60b-482d-43e2-9be1-3c2ef2b57b5a.png">

**Comment by NikRoe** [comment, marked as answer]
Hey Deki,
at the moment, you have `a`-tags that wrap around `NavLink` components that contain other `a`-tags.
When we want to work with links in NextJS, we need the `Link` component provided by NextJS. And you should not have `a`-tags nested within `a`-tags
Your current structure and code will lead to errors when trying to render your app.

Your code should look similar to this:
```js
<Div>
   <Link href="/crypto" passHref>
      <CryptoStyle>CryptoStyle></CryptoStyle>
   </Link>
</Div>
```

You can apply the same logic to your other links and have several links + `a`-tags within your wrapping container.

That wrapping container is called "Div", you should think about renaming it to something more descriptive. And it would be better so style a `nav` element instead of a `div` element, so that your HTML is more semantic.
